File: internal/analysis/testdata/branches.go
package conformance

func Count(value any, ready bool, ch chan int) int {
	if ready {
		return 1
	}
	for ready {
		break
	}
	switch value {
	case 1, 2:
	case 3:
	default:
	}
	switch value.(type) {
	case int:
	default:
	}
	select {
	case <-ch:
	default:
	}
	if ready && value != nil || ch != nil {
		return 2
	}
	return 0
}

File: internal/analysis/testdata/branches.ts
export function count(value: number, ready: boolean, values: number[]): number {
    if (ready) value++;
    for (let index = 0; index < value; index++) value--;
    for (const item of values) value += item;
    while (ready) break;
    do { value--; } while (ready);
    try { value++; } catch { value--; }
    switch (value) { case 1: break; case 2: break; default: break; }
    value = ready ? 1 : 0;
    ready = ready && value > 0 || value < -10 ?? false;
    return value;
}
